directory.go

```diff
@@ -0,0 +1,43 @@
+//  Copyright (c) 2021 Couchbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 		http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package index
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+)
+
+type Directory interface {
+	GetWriter(filePath string) (io.WriteCloser, error)
+}
+
+// FileSystemDirectory is the default implementation for the
+// Directory interface.
+type FileSystemDirectory string
```

Comment:
I think the FileSystemDirectory implementation should be in top-level bleve.  The reason is no one at this level should care, all they care about is the interface, and second, applications already import top-level bleve, while they do not likely import this package directly.